### ISSUE TITLE
Video/Popup playback for audio services (Soundcloud, Bandcamp)

### ISSUE THREAD

**Issue by opusforlife2** [TeamNewPipe/NewPipe#4105]
(Why is it that when you investigate one issue, you start discovering more and more of them!?)

1. In the current release (0.19.8) Playlist and Channel pages on audio services show the Play All and Popup buttons, and these **work**. They show the thumbnail and play just the audio. This video player and this popup player is not accessible from the main video details page. Bug, since these are audio only services. There is no use of the popup or main player.

These two buttons should not be shown on the Playlist and Channel pages if the service is an audio service.

2. In the unified player UI (post 2907), when you tap on the thumbnail on the video details page, the main player UI opens. This tap action is supposed to start background playback, like on 0.19.8. Instead, it does the same as (1). Shows a thumbnail and the main player UI. Bug, again, since these are audio only services.

Tapping on the thumbnail should start background playback, not main video playback. There is an awesome new progress indicator when playing background audio on audio services, let's promote it more. ;)

**Comment by opusforlife2** [comment]
Now I'm wondering if we could retain the main player UI for audio services. It looks neat, but the problem is that playback will stop if you have the Minimize setting set to None. Could Newpipe ignore that setting for audio services like Soundcloud and Bandcamp and just continue playing the audio?

**Comment by avently** [comment]
I always see No videos in SoundCloud tab so can't help you with this issue. Some years ago I was able to see SoundCloud but not now.

**Comment by opusforlife2** [comment]
You mean the trending tab? Or does literally no video open for you, even through search or an external link?

**Comment by avently** [comment]
Without VPN it doesn't show anything (nor trading tab, nor artist page). With VPN it shows but slow. It's probably related to network providers in Russia (but the site itself works without VPN). So it's not important

**Comment by opusforlife2** [comment]
That's quite strange that the site works but not Newpipe.

**Comment by avently** [comment]
Ok, I reread what you wrote and came to conclusion that I disagree with this issue. Let's see why:
- without background-only playback you can control playlist inside the player UI which is comfortable
- popup is not needed and it doesn't show up now, which is good. But you may say why "Play in popup" inside channel exists. For situations when someone wants to see a thumbnail with ability to select next/previous track in a queue. And it looks better (three buttons vs two). Don't need it? Don't tap on it, easy:)
- player UI allows to open an audio via browser or share the link which will not be possible in background-only playback, same for mute button

> Now I'm wondering if we could retain the main player UI for audio services. It looks neat

This is much better thought than in the first post:)

> Could Newpipe ignore that setting for audio services like Soundcloud and Bandcamp and just continue playing the audio?

Could but why? Tap on "In background" instead of a player view and you'll get your background playback. What's a problem with it?

**Comment by Stypox** [comment]
> I always see No videos in SoundCloud tab 

This happens for me too, in Italy, on my phone and on 4.4 emulator, but not on 10 emulator

**Comment by opusforlife2** [comment]
> without background-only playback you can control playlist inside the player UI which is comfortable

Agreed.

> popup is not needed and it doesn't show up now, which is good. But you may say why "Play in popup" inside channel exists. For situations when someone wants to see a thumbnail with ability to select next/previous track in a queue. And it looks better (three buttons vs two). Don't need it? Don't tap on it, easy:)

I agree. But with that logic, shouldn't popup also exist on the video details page? You could also want to play a song in a popup with auto-queue enabled.

> Could but why? Tap on "In background" instead of a player view and you'll get your background playback. What's a problem with it?

Since it's an audio only service, I think it should behave like other apps which are for streaming music, and continue playback on switching away from the app. At first I thought that there could be a separate 'Minimize' setting for audio services, but that seems more complex than just checking if the service is an audio service and continuing playback.

**Comment by wb9688** [comment]
@opusforlife2: What if you have a SC song then a YT vid queued?

**Comment by opusforlife2** [comment]
I didn't even know that was possible!

... _should_ that be possible?

**Comment by wb9688** [comment]
@opusforlife2: Yes, YT also has songs and I might want to listen to songs from different services

**Comment by opusforlife2** [comment]
This is like, the pro gamer league of Newpipe. Mixing and matching services in the quest for the ultimate queue/playlist/channel group.

**Comment by avently** [comment]
@opusforlife2 So what's the state of the issue after discussion?

**Comment by opusforlife2** [comment]
Personally, I think we should postpone discussion about this issue until after 0.20.0 has been released. This is a UI/UX discussion so it will probably take time for us to find a good solution.

---

> shouldn't popup also exist on the video details page? You could also want to play a song in a popup with auto-queue enabled.

Does anyone disagree with bringing back the popup button in audio services, though?

**Comment by opusforlife2** [comment]
Why not go the tiny step further and retain the main player UI for video services as well when playing in background?
